grassroots-backend/src/auth/Session.guard.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from "@nestjs/common";
+import { Reflector } from "@nestjs/core";
+import { DECORATOR_METADATA_INDICATING_PUBLIC_ROUTE } from "./PublicRoute.decorator";
+import { GrassrootsRequest } from "../types/GrassrootsRequest";
+import { DECORATOR_METADATA_INDICATING_OAUTH_ROUTE } from "./OAuthRoute.decorator";
+
+@Injectable()
+export class SessionGuard {
+  constructor(private reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const isRoutePublic = this.reflector.getAllAndOverride<boolean>(
+      DECORATOR_METADATA_INDICATING_PUBLIC_ROUTE,
+      [context.getHandler(), context.getClass()],
+    );
+
+    const isOAuthRoute = this.reflector.getAllAndOverride<boolean>(
```

Comment:
Is there a reason we need this to be distinct from a public route? It seems like the same logic could be achieved using `@PublicRoute`

Reply:
Done.